app/src/services/favorite/favoriteService.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+const WebtoonRepository = require("@repositories/webtoon/webtoonRepository.js");
+const UserRepository = require("@repositories/user/userRepository.js");
+const FavoriteRepository = require("@repositories/favorite/favoriteRepository.js");
+const CustomError = require("@utils/customError");
+
+class FavoriteService {
+  constructor() {
+    this.webtoonRepository = new WebtoonRepository();
+    this.userRepository = new UserRepository();
+    this.favoriteRepository = new FavoriteRepository();
+  }
+
+  async addFavorite(userId, webtoonId) {
+    const user = await this.userRepository.findById(userId);
+    if (!user) throw new CustomError("존재하지 않는 유저입니다.", 404);
+
+    const webtoon = await this.webtoonRepository.getWebtoonById(webtoonId);
+    if (!webtoon) throw new CustomError("웹툰을 찾을 수 없습니다.", 404);
+
+    const existingFavorite = await this.favoriteRepository.findFavorite(userId, webtoonId);
+    if (existingFavorite) {
+      throw new CustomError("이미 관심 목록에 추가된 웹툰입니다.", 409);
+    }
+
+    await this.favoriteRepository.addFavorite(userId, webtoonId);
+    await this.webtoonRepository.updateFavoriteCount(webtoonId, 1);
+  }
+
+  async removeFavorite(userId, webtoonId) {
+    const user = await this.userRepository.findById(userId);
+    if (!user) throw new CustomError("존재하지 않는 유저입니다.", 404);
+
+    const webtoon = await this.webtoonRepository.getWebtoonById(webtoonId);
+    if (!webtoon) throw new CustomError("웹툰을 찾을 수 없습니다.", 404);
+
+    const existingFavorite = await this.favoriteRepository.findFavorite(userId, webtoonId);
+    if (!existingFavorite) {
+      throw new CustomError("관심 목록에 없는 웹툰입니다.", 404);
+    }
+
+    await this.favoriteRepository.removeFavorite(userId, webtoonId);
+    await this.webtoonRepository.updateFavoriteCount(webtoonId, -1);
+  }
+
+  async getMyFavorites(userId, sort) {
+    const favorites = await this.favoriteRepository.findFavoritesByUserId(userId, sort);
+    return favorites;
+  }
+
+  async removeSelectedFavorites(userId, webtoonIds) {
+    const deletedWebtoonIds = await this.favoriteRepository.removeSelectedFavorites(
+      userId,
+      webtoonIds
+    );
+
+    for (const webtoonId of deletedWebtoonIds) {
+      await this.webtoonRepository.updateFavoriteCount(webtoonId, -1);
+    }
```

Comment:
지금처럼 동기적으로 이루어지지 않아도 되는 동작이라면 Promise.all을 사용하여 지연을 줄일 수 있습니다

```suggestion
    await Promise.all(deletedWebtoonIds.map((id) => this.webtoonRepository.updateFavoriteCount(webtoonId, -1))); // 에디터에서 작성한게 아니라 문법은 틀렸을 수도 있어요
```

Reply:
생각해보니까 이 작업은 꼭 동기적으로 처리할 필요가 없네요.
Promise.all을 이용해보겠습니다. 좋은 제안 감사합니다!